cmd/argo/commands/plugin/plugin.go

```diff
@@ -64,10 +65,13 @@ func loadPlugin(pluginDir string) (*loadResult, error) {
 	}
 	plug.Spec.Container.Args = []string{string(code)}
 
-	// Match default security settings for easier patch.
-	runAsNonRoot := true
-	runAsUser := int64(1000)
-	plug.Spec.Container.SecurityContext = &apiv1.SecurityContext{RunAsNonRoot: &runAsNonRoot, RunAsUser: &runAsUser}
+	if strings.Contains(pluginDir, "controller") {
+		// Match default security settings for easier patch.
+		runAsNonRoot := true
+		runAsUser := int64(1000)
+		plug.Spec.Container.SecurityContext = &apiv1.SecurityContext{RunAsNonRoot: &runAsNonRoot, RunAsUser: &runAsUser}
+	}
+
```

Comment:
Should this be its own PR? Seems separate from your example

Reply:
Fair, will do.

Reply:
https://github.com/argoproj/argo-workflows/pull/7254